sim/hunter/kill_shot.go

```diff
@@ -0,0 +1,79 @@
+package hunter
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+)
+
+func (hunter *Hunter) registerKillShotSpell() {
+	if hunter.HasPrimeGlyph(proto.HunterPrimeGlyph_GlyphOfKillShot) {
```

Comment:
when doing registers based on a condition we usually prefer the Early Exit Pattern:
```
func (hunter *Hunter) registerKillShotSpell() {
	if hunter.HasPrimeGlyph(proto.HunterPrimeGlyph_GlyphOfKillShot) {
        	return;
	}
        spellImpl...
```

Reply:
ahh i just saw this is an extra condition on the spell and not the spell itself you can ignore